util/cache_allocator.h

```diff
@@ -0,0 +1,38 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+
+#pragma once
+
+#include "rocksdb/table.h"
+
+namespace rocksdb {
+
+struct CustomDeleter {
+  CustomDeleter(CacheAllocator* a = nullptr) : allocator(a) {}
+
+  void operator()(char* ptr) const {
+    if (allocator) {
+      allocator->deallocate((void*)ptr);
+    } else {
+      delete ptr;
```

Comment:
This will cause pointers allocated with `new char[]` to be deleted using `delete` rather than `delete[]`, which is incorrect. You probably want `delete[] ptr`.